src/AppRun.c

```diff
@@ -25,182 +25,298 @@ THE SOFTWARE.
 #include <stdlib.h>
-#include <unistd.h>
-#include <libgen.h>
-#include <dirent.h>
 #include <string.h>
-#include <errno.h>
+#include <unistd.h>
 
-#define die(...)                                    \
-    do {                                            \
-        fprintf(stderr, "Error: " __VA_ARGS__);     \
-        exit(1);                                    \
-    } while(0);
-#define SET_NEW_ENV(str,len,fmt,...)                \
-    format = fmt;                                   \
-    length = strlen(format) + (len);                \
-    char *str = calloc(length, sizeof(char));     \
-    snprintf(str, length, format, __VA_ARGS__);   \
+/** Macro to set environment variables. */
+#define SET_NEW_ENV(str, len, fmt, ...)         \
+    format = fmt;                               \
+    length = strlen(format);                    \
+    char* str = calloc(length, sizeof(char));   \
+    snprintf(str, length, format, __VA_ARGS__);  \
     putenv(str);
-#define MAX(a,b)    (a > b ? a : b)
-#define bool int
-#define false 0
-#define true -1
 
+/** Macro to return the largest of a pair of numbers */
+#define MAX(a, b) (a > b? a : b)
+
+// TODO: Is this arbitrary? Check and refine.
 #define LINE_SIZE 255
 
-int filter(const struct dirent *dir) {
-    char *p = (char*) &dir->d_name;
-    p = strrchr(p, '.');
-    return p && !strcmp(p, ".desktop");
+/** Throw error and exit. */
+void die(const char * format, ...) {
+    va_list args;
+    va_start(args, format);
+    vprintf(format, args);
+    va_end(args);
+    exit(1);
 }
 
+/** Check for a .desktop file at the given path.
+  * \param the file path to check
+  * \return true if the path points to a .desktop file, else false.
+  */
+ int filter(const struct dirent* dir) {
+     // Get pointer to file path.
+     char* p = (char*) &dir->d_name;
+     // Read only the file extension (start reading from last dot in path)
+     p = strrchr(p, '.');
+     // Return true if the file extension .desktop is found.
+     return (p && !strcmp(p, ".desktop")) ? 1 : 0;
+ }
+
 int main(int argc, char *argv[]) {
-    char *appdir = dirname(realpath("/proc/self/exe", NULL));
-    if (!appdir)
+    // Get path to the application directory.
+    char* appdir = dirname(realpath("/proc/self/exe", NULL));
+    /* If there's a problem accessing the application directory,
+     * throw fatal error.
+     */
+    if (!appdir) {
         die("Could not access /proc/self/exe\n");
+    }
 
+    // Check for .desktop files
+    // The return code for whether we found a .desktop file
     int ret;
-    struct dirent **namelist;
+    // An array of pointers to lists of file paths.
+    struct dirent** namelist;
+    // Scan for .desktop files.
     ret = scandir(appdir, &namelist, filter, NULL);
 
     if (ret == 0) {
-        die("No .desktop files found\n");
+        // If we found no .desktop files, there's nothing to run. Fatal error.
+        die("No .desktop files found.\n");
     } else if(ret == -1) {
```

Comment:
Would appreciate if you could add the missing space here. Spaces are probably also missing in other places.